packages/burner-plugin/src/ui/MoviePage.tsx

```diff
@@ -31,91 +55,177 @@ const MoviePage: React.FC<MoviePageContext> = ({
   const Page = burnerComponents.Page as WhiteRabbitPageType;
 
   const { pathname } = location;
-  const [, tokenId] = pathname.match(/\/movie\/(\d+)\/?/i) || [];
-  let imdbId: string = '';
+  const [, tokenId, region, windowV, pss] =
+    pathname.match(/\/movie\/(\d+)\/([^\/]+)\/(\d+)\/([^\/]+)\/?/i) || [];
+  let imdbId: string = "";
   try {
     imdbId = utils.tokenToImdb(tokenId);
   } catch (e) {
-    console.error('Cannot decode tokenId', e);
+    console.error("Cannot decode tokenId", e);
   }
 
   const [account] = accounts;
   const asset = assets[0];
   const value = plugin.getPaymentAmount();
 
   if (!imdbId) {
-    return (
-      <MessagePage>
-        Invalid movie token Id
-      </MessagePage>
-    );
+    return <MessagePage>Invalid movie token Id</MessagePage>;
   }
-  
+
   useEffect(() => {
+    const convertArrayToObject = (array: Array<any>, key: string) => {
+      return array.reduce((obj: any, item: any) => ({
+        ...obj,
+        [item[key]]: item,
+      }), {});
+    };
+
+    const getDepositEvents = async (): Promise<any> => {
+      let wrEvents: any = {};
+      if (typeof Storage !== "undefined") {
+        wrEvents =
+          window.localStorage.getItem("wr-events") !== null
+            ? JSON.parse(window.localStorage.getItem("wr-events") as any)
+            : {};
+      }
+      const startingBlock =
+        "blockNumber" in wrEvents ? wrEvents["blockNumber"] : 0;
```

Comment:
more succinct and idiomatic IMO:

```suggestion
        wrEvents.blockNumber || 0;
```